crates/oxc_transformer/src/plugins/module_runner_transform.rs

```diff
@@ -0,0 +1,2055 @@
+    (0, __vite_ssr_import_0__.fn)();
+  }
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_when_variable_in_scope() {
+        test_same(
+            "import { fn } from 'vue';function A(){ const fn = () => {}; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  const fn = () => {};
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_destructuring_object() {
+        test_same(
+            "import { fn } from 'vue';function A(){ let {fn, test} = {fn: 'foo', test: 'bar'}; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let { fn, test } = {
+    fn: 'foo',
+    test: 'bar'
+  };
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_destructuring_array() {
+        test_same(
+            "import { fn } from 'vue';function A(){ let [fn, test] = ['foo', 'bar']; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let [fn, test] = ['foo', 'bar'];
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_catch_clause() {
+        test_same(
+            "import {error} from './dependency';try {} catch(error) {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['error'] });
+try {} catch (error) {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_imported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; class A extends Foo {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['Foo'] });
+class A extends __vite_ssr_import_0__.Foo {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_exported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; export default class A extends Foo {} export class B extends Foo {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['Foo'] });
+class A extends __vite_ssr_import_0__.Foo {}
+Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return A;
+  }
+});
+class B extends __vite_ssr_import_0__.Foo {}
+Object.defineProperty(__vite_ssr_exports__, 'B', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return B;
+  }
+});",
+        );
+    }
+
+    /// <https://github.com/vitejs/vite/issues/4049>
+    #[test]
+    fn should_handle_default_export_variants() {
+        // default anonymous functions
+        test_same(
+            "export default function() {}",
+            "Object.defineProperty(__vite_ssr_exports__, 'default', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return function() {};
+  }
+});",
+        );
```

Comment:
I think this should be
```js
const temp = function () {};
Object.defineProperty(__vite_ssr_exports__, 'default', {
  enumerable: true,
  configurable: true,
  get() {
    return temp;
  }
});
// or
Object.defineProperty(__vite_ssr_exports__, 'default', {
  enumerable: true,
  configurable: true,
  value: function () {}
});
// or
__vite_ssr_exports__.default = function () {};
```
otherwise
```js
// foo.js
import bar from './bar.js'
import baz from './baz.js'
if (bar !== baz) {
  throw new Error("should be same reference")
}
// bar.js
import baz from './baz.js'
export default baz
// baz.js
export default function () {}
```
would probably fail.
(same for anonymous classes)

Reply:
Fixed in https://github.com/oxc-project/oxc/pull/9371/commits/328009e5415b84c2fd8d5624da8d4905a718feb5